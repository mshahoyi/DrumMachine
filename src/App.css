#centerDiv {
}

#buttonArea {
  width: 50%;
  display: inline-block;
  text-align: center;
}

#buttonArea button {
  width: 50px;
  height: 50px;
  margin: 5px;
}

#settingArea {
  float: right;
  width: 50%;
  display: inline-block;
}