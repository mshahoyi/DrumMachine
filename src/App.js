import React, { Component } from 'react';
import './App.css';
import Button from './Button.js'
import Switch from './Switch.js';
import Display from './Display';
import Volume from './Volume.js'

const STATUS = {
  MODE: {
    PIANO: 'PIANO',
    HEATER: 'HEATER'
  }
}

const CHORD_TO_FILE = {
  A: '5th_String_A.mp3',
  D: '4th_String_D.mp3'
}

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      power: true,
      type: STATUS.MODE.PIANO,
      volume: 30,
      display: 'welcome'
    }
    
    this._power_function = this._power_function.bind(this);
    this._mode_changed = this._mode_changed.bind(this);
    this._handle_volume = this._handle_volume.bind(this);
  }

  render() {
    return (
      <div id='centerDiv'>
        <div id='buttonArea'>
          <div id='row1'>
            <Button text='A' soundPath={this.soundPath('A')} power={this.state.power} volume={this.state.volume/30}/>
            <Button text='Am' />
            <Button text='D' soundPath={this.soundPath('D')} power={this.state.power} volume={this.state.volume/30}/>
          </div>
          <div id="row2">
            <Button text='Dm' />
            <Button text='E' />
            <Button text='Em' />
          </div>
          <div id='row3'>
            <Button text='G' />
            <Button text='C' />
            <Button text='F' />
          </div>
        </div>
        <div id='settingArea'>
          <Switch name={'Power'} clickfn={this._power_function} state={this.state.power}/>
          <Display text={this.state.display} animation={'display-animation'}/>
          <Volume handleChange={this._handle_volume} volume={this.state.volume}/>
          <Switch state={this.modeToBool(this.state.type)} name={'Mode'} clickfn={this._mode_changed}/>
        </div>
      </div>
    );
  }

  _power_function() {
    this.setState((state) => {
      return {
        power: !state.power,
        display: 'Power ' + (!state.power ? 'on' : 'off')
      }
    })
  }

  _mode_changed() {
    this.setState(state => {
      let nextMode = this.toggleMode(state.type);
      return {
        type: nextMode,
        display: nextMode
      }
    })
  }

  modeToBool = (string) => { 
    if(string === STATUS.MODE.PIANO) return true;
    else return false;
  }

  toggleMode(mode) {
    if(mode === STATUS.MODE.PIANO) return STATUS.MODE.HEATER
    else return STATUS.MODE.PIANO
  }

  soundPath(chord) {
    let path = './Chords/' + CHORD_TO_FILE[chord];
    return path;
  }

  _handle_volume(volume) {
    this.setState({volume: volume, display: 'Volume: ' + volume})
  }
}

export default App;
