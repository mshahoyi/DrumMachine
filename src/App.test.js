import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { render, cleanup, getByText } from 'react-testing-library';
import renderer from 'react-test-renderer';

it('app should have a p element div inside', () => {
  const tree = renderer.create.toJSON();
  expect(tree).toMatchSnapshot();
});
