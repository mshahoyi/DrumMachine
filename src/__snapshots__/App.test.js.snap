// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should have a p element div inside 1`] = `
<div>
  me
</div>
`;
